JavaScript/homeword_assignments/deepFreeze.js

```diff
@@ -0,0 +1,32 @@
+Object.prototype.deepFreeze = function (obj){
+    Object.freeze(obj);
+
+    for(const key of Object.keys(obj)){
+        if(typeof obj[key] === 'object' && obj[key] != null){
```

Comment:
@goyal-Dushi, please use !== instead of just != in the case of JS which also takes care of data type